pkg/beyla/os.go

```diff
@@ -21,3 +25,146 @@ func CheckOSSupport() error {
 	}
 	return nil
 }
+
+type osCapability uint8
+
+type capDesc struct {
+	osCap osCapability
+	str   string
+
+	// kernMaj.kernMin is the MAXIMUM kernel version this capability is needed
+	// if push comes to shove in the future, we may want to implement proper
+	// ranges here
```

Comment:
The only known capability we required that is tied to a kernel version range is `CAP_SYS_RESOURCE`, which needs to be present for kernels < 5.11.

Originally I had an explicit branch to handle this particular case, but I decided to generalise it at the expense of having to specify `kernMaj` and `kernMin` for every capability we require. Perhaps there's a better way?